sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -359,11 +359,23 @@ class DeviceBinaryImage {
     return Format;
   }
 
-  /// Gets the iterator range over specialization constants in this this binary
-  /// image. For each property pointed to by an iterator within the range, the
-  /// name of the property is the specializaion constant symbolic ID and the
-  /// value is 32-bit unsigned integer ID.
-  const PropertyRange &getSpecConstants() const { return SpecConstIDMap; }
+  /// Gets the iterator range over scalar specialization constants in this this
+  /// binary image. For each property pointed to by an iterator within the
+  /// range, the name of the property is the specializaion constant symbolic ID
+  /// and the value is 32-bit unsigned integer ID.
+  const PropertyRange &getScalarSpecConstants() const {
+    return ScalarSpecConstIDMap;
+  }
+  /// Gets the iterator range over composite specialization constants in this
+  /// this binary image. For each property pointed to by an iterator within the
+  /// range, the name of the property is the specializaion constant symbolic ID
+  /// and the value is a list of tuples of 32-bit unsigned integer values, which
+  /// encode scalar specialization constants, that form a composite one.
```

Comment:
```suggestion
  /// encode scalar specialization constants, that form the composite one.
```

What if it is a nested structural type? Can you maybe give an example here?

Reply:
The whole structure will be flattened. Sure, I will add an example

Reply:
Fixed and added an example in 0800c2bc8278c364eeee5951d0be180df228dcef